workflows/syncing-production-data-to-qa.md

```diff
@@ -0,0 +1,135 @@
+```
+
+**3. Logstash config**
+
+Build an appropriate `logstash.conf`, such as:
+```
+input {
+  elasticsearch {
+    hosts => ["https://[fqdn of production domain]:443"]
+    index => "resources-2018-04-09"
+    docinfo => true
+    query => '{ "sort": [ "uri" ], "query": { "query_string": { "query": "uri:>0" } } }'
+  }
+}
+filter {
+  mutate {
+    remove_field => [ "@version", "@timestamp", "publisherLiteral" ]
+  }
+}
+output {
+  elasticsearch{
+    hosts => ["https://[fqdn of qa domain:443"]
+    manage_template => false
+    index => "resources-[YYYY-MM-DD]"
+    document_type => "%{[@metadata][_type]}"
+    document_id => "%{[@metadata][_id]}"
+  }
+}
+```
+
+Note the input/output may need to be changed from "opensearch" to "elasticsearch" depending on whether the domain is actually AWS OpenSearch or a classic Elasticsearch domain. The 5.3 domains appear to be usable as "elasticsearch" targets even though AWS calls them opensearch - probably grandfathered access points.
+
+Note also inclusion of "parallelPublisher" in `remove_field` to tell logstash to remove parallelPublisher when found. This was a deprecated property in favor of parallelPublisherLiteral property in any document it's 
+
+**4. Begin the copy**
+
+Newer versions of logstash appear to drop support for older Elasticsearch domains. See [this support matrix](https://www.elastic.co/support/matrix#matrix_compatibility). I downloaded [logstash 5.6 from here](https://www.elastic.co/downloads/past-releases/logstash-5-6-5) for use with a couple ES 5.3 domains. For newer Elasticsearch versions, you may be able to just `brew install logstash`. The [complete set of options to logstash are here](https://www.elastic.co/guide/en/logstash/current/running-logstash-command-line.html)
+
+Execute the job:
+
+```
+LS_JAVA_OPTS="-Xms256m -Xmx2048m" logstash -f logstash.conf
+```
+
+Note: If logstash reports You may need to install opensearch input/output plugins:
+```
+cd /usr/local/Cellar/logstash/8.13.2/bin # Change to logstash path
+./logstash-plugin install logstash-input-opensearch
+```
+
+The `LS_JAVA_OPTS` bit above invites logstash to use a reasonable amount of memory for the work it needs to do rather than fall over with a `java.lang.OutOfMemoryError` around about 7M records. A proverbial "make yourself at home" to the JVM.
+
+Follow progress by checking index doc count via `GET https://[fqdn of qa domain]]/_cat/indices?v`
+
+**If the pipeline fails**, you may need to kill the process via multiple `CTRL-c`s. Resume from the last indexed `uri`. Determine the last seen `uri` by POSTing the following to `{{ES_BASE_URI}}/{{ES_INDEX_NAME}}/resource/_search`
+
+```
+{
+  "size": 1,
+  "sort" : [
+    { "uri": "desc" }
+  ]
+}
+```
+
+The single record returned has the last indexed `uri` (which is also the `_id` value). Once obtained, cause `logstash` to begin where you left off by editing the `query` line in your logstash conf as follows:
```

Comment:
can you move the first sentence of this line above the query body? i gleaned this was the case before reading it, but if you're going to say it, you should probably say it before the body.